src/logic/calculate.js

```diff
@@ -125,6 +125,20 @@ export default function calculate(obj, buttonName) {
     return { operation: buttonName };
   }
 
+  if (buttonName === "←") { // Backspace functionality
+    if (obj.next) {
+      return { next: obj.next.slice(0, -1) };
+    }
+    return {};
+  }
+
+  if (buttonName === "√") { // Square root functionality
+    if (obj.next) {
+      return { next: Big(obj.next).sqrt().toString() };
+    }
+    return {};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The square root functionality also looks good. But it doesn't handle the case when `obj.next` is a negative number. The square root of a negative number is not a real number and should be handled appropriately. 

```diff
  if (buttonName === "√") { // Square root functionality
    if (obj.next && Number(obj.next) >= 0) {
      return { next: Big(obj.next).sqrt().toString() };
    }
    return { error: "Invalid input for square root" };
  }
```
Please note that this will introduce a new 'error' field in the returned object. You need to handle this 'error' field wherever this function is being used.


<!-- This is an auto-generated comment by OSS CodeRabbit -->